sdk/cosmos/azure_data_cosmos/tests/framework/test_account.rs

```diff
@@ -0,0 +1,181 @@
+        options: Option<TestAccountOptions>,
+    ) -> Result<Self, Box<dyn std::error::Error>> {
+        let options = options.unwrap_or_default();
+        let splat = connection_string.split(';');
+        let mut account_endpoint = None;
+        let mut account_key = None;
+        for part in splat {
+            let part = part.trim();
+            if part.is_empty() {
+                continue;
+            }
+
+            let (key, value) = part.split_once('=').ok_or("invalid connection string")?;
+            match key {
+                "AccountEndpoint" => account_endpoint = Some(value.to_string()),
+                "AccountKey" => account_key = Some(Secret::new(value.to_string())),
+                _ => {}
+            }
+        }
+
+        let Some(endpoint) = account_endpoint else {
+            return Err("invalid connection string, missing 'AccountEndpoint'".into());
+        };
+
+        let Some(key) = account_key else {
+            return Err("invalid connection string, missing 'AccountKey'".into());
+        };
+
+        let context_id = format!(
+            "{}_{}_{}",
+            context.test_name(),
+            OffsetDateTime::now_utc().format(format_description!(
+                "[year]_[month]_[day]T[hour]_[minute]_[second]"
+            ))?,
+            Uuid::new_v4().as_simple()
+        );
+
+        TRACING.call_once(|| {
+            // Enable tracing for tests, if it's not already enabled
+            _ = tracing_subscriber::fmt()
+                .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+                .try_init();
+
+            // Ignore the failure. The most likely failure is that a global default trace dispatcher has already been set.
+            // And we don't want the tracing support to bring down the tests.
+        });
+
+        Ok(TestAccount {
+            context_id,
+            endpoint,
+            key,
+            options,
+        })
+    }
+
+    /// Create a [`CosmosClient`](azure_data_cosmos::CosmosClient) that connects to this account using a connection string.
+    #[cfg(feature = "key_auth")]
+    pub fn connect_with_key(
+        &self,
+        mut options: Option<CosmosClientOptions>,
+    ) -> Result<azure_data_cosmos::CosmosClient, Box<dyn std::error::Error>> {
+        let allow_invalid_certificates = match self.options.allow_invalid_certificates {
+            Some(b) => b,
+            None => std::env::var(ALLOW_INVALID_CERTS_ENV_VAR).map(|s| s.parse())??,
+        };
+
+        if allow_invalid_certificates {
+            let client = ClientBuilder::new()
+                .danger_accept_invalid_certs(true)
+                .pool_max_idle_per_host(0)
+                .build()?;
+            options = {
+                let mut o = options.unwrap_or_default();
+                o.client_options
+                    .set_transport(TransportOptions::new(Arc::new(client)));
+                Some(o)
+            };
+        }
```

Comment:
Calling some special attention here, since I'm calling `danger_accept_invalid_certs`.

The emulator uses a self-signed certificate. I tried putting the cert in the root store, but something in the pipeline is still unhappy with that _specifically because the certificate is self-signed_ (i.e. even if it's trusted and `curl` and `openssl s_client` are both happy). So, we give the option to disable cert validation _in these tests only_ (notice that there's no built-in option to do this in the SDK itself, it's a custom transport used by the tests). The flag to bypass cert validity checks is set in **either** of these circumstances:

* `AZURE_COSMOS_CONNECTION_STRING=emulator` - The user is explicitly using the emulator, thus we know we'll have a self-signed cert.
* `AZURE_COSMOS_ALLOW_INVALID_CERT=true` - The user has explicitly requested that we allow invalid certs [^1].

[^1]: Again, I want to make clear that it's the _test_ reading this value. This environment variable is meaningless to the SDK and won't disable any checks in the SDK itself.

Reply:
In fact, `rustls` specifically disallows self-signed certificates **even if they are trusted roots**, unless you set `danger_accept_invalid_certs`:

>> For reasons explained in the other sections of this manual, rustls does not and will not support: 
>> * ...
>> * Using CA certificates directly to authenticate a server/client (often called “self-signed certificates”). Rustls’ default certificate verifier does not support using a trust anchor as both a CA certificate and an end-entity certificate in order to limit complexity and risk in path building. While dangerous, all authentication can be turned off if required – see the [example code](https://github.com/rustls/rustls/blob/992e2364a006b2e84a8cf6a7c3eaf0bdb773c9de/examples/src/bin/tlsclient-mio.rs#L318) *
>
> https://docs.rs/rustls/latest/rustls/manual/_04_features/index.html#non-features

Reply:
Quick question here - for the new emulator, we are moving towards using `http` instead of `https` by default, so we can get out of the self-signed cert business a bit. Would anything need to change here to be able to use that?

Reply:
Using `http` for the emulator would certainly be preferable as it makes clear the security limitations (i.e. the emulator being intended strictly for development/testing on your local machine). I don't think anything blocks us from using `http` in our SDK. We expect a full URL, scheme included, everywhere, so if you specify `http`, we'll use `http`.

Does the current emulator support `http` at all?

Reply:
> Does the current emulator support http at all?

Sadly, no.